internal/controller/replication.storage/status.go

```diff
@@ -74,26 +82,30 @@ func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGenerat
 }
 
 // sets conditions when volume promotion was failed due to failed validation.
-func setFailedValidationCondition(conditions *[]metav1.Condition, observedGeneration int64) {
+func setFailedValidationCondition(conditions *[]metav1.Condition, observedGeneration int64, message, messageFromDriver string) {
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            v1alpha1.MessageFailedPromoted,
 		Type:               v1alpha1.ConditionCompleted,
 		Reason:             v1alpha1.FailedToPromote,
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionFalse,
 	})
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            message,
 		Type:               v1alpha1.ConditionDegraded,
 		Reason:             v1alpha1.Error,
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionTrue,
 	})
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            v1alpha1.MessageNotResyncing,
 		Type:               v1alpha1.ConditionResyncing,
 		Reason:             v1alpha1.NotResyncing,
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionFalse,
 	})
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            fmt.Sprintf("%s: %s", v1alpha1.MessageFailedPreCondition, messageFromDriver),
```

Comment:
it can't be for same conditions, cause there are cases like for failed promote, the message will contain the reason for failure to promote, while in case of resync, it will printed in resync part.  based on which message belongs where, I have added it.